src/Compilers/CSharp/Portable/Symbols/Tuples/TupleTypeSymbol.cs

```diff
@@ -109,7 +109,7 @@ internal sealed class TupleTypeSymbol : WrappedNamedTypeSymbol
             }
 
             var constructedType = Create(underlyingType, elementNames, errorPositions, locationOpt, elementLocations);
-            if (shouldCheckConstraints)
+            if (shouldCheckConstraints && diagnostics != null)
```

Comment:
I guess it is fine to suppress constraints checks as well when the type is not going to be used at the end.

---
In reply to: [123330421](https://github.com/dotnet/roslyn/pull/20295#discussion_r123330421) [](ancestors = 123330421)